src/CursorBatching.ts

```diff
@@ -71,26 +71,26 @@ export default class CursorBatching {
     }
   }
 
-  private async publishFromBuffer(bufferName: keyof OutgoingBuffers, channelName: string) {
+  private async publishFromBuffer(bufferName: keyof OutgoingBuffers, eventName: string) {
     if (!this.isRunning) {
       this.isRunning = true;
-      await this.batchToChannel(bufferName, channelName);
+      await this.batchToChannel(bufferName, eventName);
     }
   }
 
-  private async batchToChannel(bufferName: keyof OutgoingBuffers, channelName: string) {
+  private async batchToChannel(bufferName: keyof OutgoingBuffers, eventName: string) {
     if (!this.has.movement) {
       this.isRunning = false;
```

Comment:
isRunning will be set to false here and the function will return early when there is no movement without considering whether or not this is a movement batch or not

Reply:
Ah yeah, good catch. I'm not sure about this setup anyway. Only sending the relevant data to the event being sent means we can't rewind the channel and get the full cursor (for 'getAll' for example). Sending the full cursor means we need to keep the events together and fire both events with all the most recent data attached, but what data seems a bit odd. I'm setting this back to draft/in progress because I need to think about it.
In part, I'm not sure how to sort that out because I'm not really sure what the pattern would be to use a channel's rewind functionality for something like this.